src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java

```diff
@@ -0,0 +1,231 @@
+        return solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORONE)
+            && caseData.getRespondent1ResponseDate() != null;
+    }
+
+    private boolean isResponseSubmissionPassedDeadline(CallbackParams callbackParams,
+                                                       CaseData caseData,
+                                                       LocalDateTime dateTime) {
+
+        var respondent1ResponseDeadline = caseData.getRespondent1ResponseDeadline();
+        var respondent2ResponseDeadline = caseData.getRespondent2ResponseDeadline();
+
+        return (isRespondent1PassedDeadline(caseData, callbackParams, dateTime, respondent1ResponseDeadline)
+            || isRespondent2PassedDeadline(caseData, callbackParams, dateTime, respondent2ResponseDeadline));
+    }
+
+    private boolean isRespondent1PassedDeadline(CaseData caseData,
+                                                CallbackParams callbackParams,
+                                                LocalDateTime dateTime,
+                                                LocalDateTime respondent1ResponseDeadline) {
+        return solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORONE)
+            && caseData.getRespondent1ResponseDate() == null
+            && respondent1ResponseDeadline != null
+            && dateTime.toLocalDate().isAfter(respondent1ResponseDeadline.toLocalDate());
+    }
+
+    private boolean isRespondent2PassedDeadline(CaseData caseData,
+                                                CallbackParams callbackParams,
+                                                LocalDateTime dateTime,
+                                                LocalDateTime respondent2ResponseDeadline) {
+        return solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORTWO)
+            && caseData.getRespondent2ResponseDate() == null
+            && respondent2ResponseDeadline != null
+            && dateTime.toLocalDate().isAfter(respondent2ResponseDeadline.toLocalDate());
+    }
+
+    private boolean solicitorRepresentsOnlyOneOrBothRespondents(CallbackParams callbackParams, CaseRole caseRole) {
+        CaseData caseData = callbackParams.getCaseData();
+        UserInfo userInfo = userService.getUserInfo(callbackParams.getParams().get(BEARER_TOKEN).toString());
+        return stateFlowEngine.evaluate(caseData).isFlagSet(TWO_RESPONDENT_REPRESENTATIVES)
+            && coreCaseUserService.userHasCaseRole(
+            caseData.getCcdCaseReference().toString(),
+            userInfo.getUid(),
+            caseRole
+        );
+    }
+
+    private boolean solicitorRepresentsOnlyOneOfRespondents(CallbackParams callbackParams, CaseRole caseRole) {
+        return solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, caseRole);
+    }
+
+    private List<LocationRefData> fetchLocationData(CallbackParams callbackParams) {
```

Comment:
again, fetch can be used but the standard is getSomething, setSomething, isSomething